src/client/automation/playback/click/index.js

```diff
@@ -222,21 +221,43 @@ export default class ClickAutomation {
         return focusAndSetSelection(elementForFocus, simulateFocus, this.caretPos);
     }
 
+    _getElementForClick (targetElement, topElement, targetElementParentNodes, topElementParentNodes) {
+        var areElementsSame = domUtils.isTheSameNode(topElement, targetElement);
+
+        // NOTE: Mozilla Firefox always skips click, if an element under cursor has been changed after mousedown.
+        if (browserUtils.isFirefox)
+            return areElementsSame ? targetElement : null;
+
+        if (!areElementsSame) {
+            if (targetElement.contains(topElement) && !domUtils.isEditableFormElement(topElement))
+                return targetElement;
+
+            if (topElement.contains(targetElement))
+                return topElement;
+
+            // NOTE: If elements are not in parent relationship, non-ff browsers
+            // are raising`click` event for their common parent.
+            var commonElement = arrayUtils.getCommonElement(topElementParentNodes, targetElementParentNodes);
+
+            if (!commonElement)
```

Comment:
What situation it's possible in? I suppose they should have at least one common element - body